Sync/Synchronizers/ClientsSynchronizer.swift

```diff
@@ -33,12 +46,42 @@ public class WipeCommand: Command {
     public func run(synchronizer: ClientsSynchronizer) -> Success {
         return succeed()
     }
+
+    public func syncCommand() -> SyncCommand? {
```

Comment:
If you really need to pair up a client GUID and a command for a while, and you want that to be `SyncCommand`, this is the place to do it:

```
public func syncCommandForClientGUID(guid: GUID) -> SyncCommand {
```

Making `SyncCommand` generic enough to always represent every `Command` allows you to get rid of the optional result, too.